### DISCUSSION TITLE
Is there a way to delete all authenticated users?

### DISCUSSION THREAD

**Discussion by jollygreenegiant** [supabase/supabase#13836] [Questions]
I have a test project set up, and migrated a bunch of users over from Firebase to test the new version of my app with.  I'd like to be able to delete all those auth users and start fresh with some new function triggers I've made.

Is there a way to do this?  I've tried truncating and deleting all records from auth.users, but keep running into issues with it have foreign key relations that stop records inside it from being deleted.  I know I can manually delete users from the dashboard, but there are too many users in there for me to be able to delete all of them manually.

**Comment by GaryAustin1** [comment]
You could write some code to list users and then loop deleting them using the API with service_role key from a safe environment....

https://supabase.com/docs/reference/javascript/admin-api

Other than that you have to deal with several tables to clear all user info and probably look at gotrue code to insure you truncate them in the correct order.  I've not seen anyone with code to do it.
